content/docs/command-reference/exp/run.md

```diff
@@ -35,27 +35,24 @@ data and code updates, or <abbr>hyperparameter</abbr> tuning. You can use the
 `--set-param` option to change `dvc param` values on-the fly.
 
 Each `dvc exp run` creates a variation based on the latest project version
-committed to Git, and tracks this experiment automatically with an automatic
+committed to Git (`HEAD`), and tracks this experiment with an auto-generated
 name like `exp-bfe64` (which can be customized with the `--name` option).
 
 <details>
 
 ### How does DVC track experiments?
 
-Internally, `dvc exp` uses actual commits under custom
+Experiments are custom
 [Git references](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
-(found in `.git/refs/exps`). Each commit has the Git `HEAD` as parent and has
-it's own SHA-256 hash. These are not pushed to the Git remote by default (see
+(found in `.git/refs/exps`) with a commit whose parent is the `HEAD` at the time
+of running. Note that they're not pushed to the Git remote by default (see
```

Comment:
Not sure if "at the time of running" needs to be clarified here. Technically it is more like "at the time the experiment is queued". (For regular runs without `--queue/--run-all` it's just that we queue the single exp and execute it immediately afterwards.)

The important thing is that if `HEAD` has moved between `exp run --queue`, and `exp run --run-all`, we still execute each queued experiment based on `HEAD` at the time of `exp run --queue`.

Reply:
OK yeah we can omit that detail (done). But it's not clear that the exp commit doesn't become HEAD (i.e. doesn't get checked out) ⌛

Reply:
OK I clarified that too.

> if HEAD has moved between exp run --queue, and exp run --run-all, we still execute each queued experiment based on HEAD

- [ ] I tried testing this but it gets complicated... May require further QA/doc in the future. Is it a use case though? Queueing experiments from different HEADs and then running them all later. Thanks @pmrowla

Reply:
It's intended to work that way, but I'm not sure how common of a use case it would be for users.

Reply:
I would expect my queued experiments to reflect the time they were queued. I'm not sure how common it would be to change HEAD before running experiments in the queue. It might be useful to queue experiments from different branches. It also might work as an alternative to making multiple copies of a repo in cases where someone wants to run different stages simultaneously (`dvc exp run --queue stage1 && dvc exp run --queue stage2 && dvc exp run --run-all -j 2`). I'd be surprised if people use it this way, though.